src/psyclone/tests/psyir/tools/dependency_tools_test.py

```diff
@@ -729,3 +736,305 @@ def test_gocean_parallel():
             "'u_fld(i,j - 1)' in '< kern call: stencil_not_parallel_code >' "
             "are dependent and cannot be parallelised. Variable: 'u_fld'."
             in str(dep_tools.get_all_messages()[0]))
+
+
+def test_fuse_different_variables_with_access(fortran_reader):
+    '''Test that fusing loops with different variables is disallowed when
+    either loop uses the other loops variable for any reason.'''
+    code = '''subroutine sub()
+    integer :: ji, jj, n, jk
+    integer, dimension(10, 10) :: s, t
+    do jj = 1, n
+      do ji = 1, 10
+        s(ji, jj) = t(ji, jj) + 1
+      end do
+      do jk = 1, 10
+        ji = jk
+        s(jk, jj) = t(jk, jj) + ji
+      end do
+    end do
+    end subroutine sub'''
+    psyir = fortran_reader.psyir_from_source(code)
+    loops = psyir.children[0].walk(Loop)
+    dep_tools = DependencyTools()
+
+    assert not dep_tools.can_loops_be_fused(loops[1], loops[2])
+    assert len(dep_tools.get_all_messages()) == 1
+    msg = dep_tools.get_all_messages()[0]
+    assert ("Second loop contains accesses to the first loop's variable: ji"
+            in str(msg))
+    assert msg.var_names[0] == "ji"
+
+    assert not dep_tools.can_loops_be_fused(loops[2], loops[1])
+    msg = dep_tools.get_all_messages()[0]
+    assert ("First loop contains accesses to the second loop's variable: ji"
+            in str(msg))
+    assert msg.var_names[0] == "ji"
+
+
+# ----------------------------------------------------------------------------
+def test_fuse_independent_array_stencil(fortran_reader):
+    '''Test that using arrays which are not dependent on the loop variable
```

Comment:
This docstring and the comment below don't seem to match what is tested. Does the test title need changing too?